### PR TITLE
Add custom spec description option to `run_test!`

### PR DESCRIPTION
**Pull request by mattpolito**
## Problem
Specifications generated by `run_test!` are essentially a hard coded string of "returns a RESPONSE_CODE response". If you need to customize this you have to break down into the individual parts that `run_test!` constructs for us.

```ruby
it "custom specification description here" do |example|
  submit_request(example.metadata)
  assert_response_matches_metadata(example.metadata)
end
```

## Solution
To facilitate configuration without the need to break out of the nicety that `run_test!` provides, we can now pass a `description` as the first argument. 

`run_test!("custom specification description here")`

A huge benefit to this is that when paired with auto generation of request/response examples, many example scenarios can be provided for documentation within the same `response` grouping.

NOTE: `description` is not an allowable option to pass down to a specification, so we delete it vs forwarding down to `it`.

### The changes I made are compatible with:
- [x] OAS2
- [x] OAS3
- [x] OAS3.1

### Checklist
- [x] Added tests
- [x] Changelog updated
- [x] Added documentation to README.md
- [x] Added example of using the enhancement into [test-app](https://github.com/rswag/rswag/tree/master/test-app)

### CODE REVIEW

**Comment by mattpolito** [comment]
@romanblanco If rswag is in need of maintainer help. We can be of assistance.

**Comment by a-lavis** [comment]
This is a great idea!  I have a proposal of how we could do it slightly differently here: https://github.com/hashrocket/rswag/pull/1
Let me know your thoughts!

**Comment by mattpolito** [comment]
Updated PR with test-app scenario

**Comment by BookOfGreg** [comment]
Hi @mattpolito, Roman pointed out your PR here on this project and I've chosen to grant you Write access to the project.
We hope you and the rest of the team at Hashrocket a good time with rswag.

**Comment by mattpolito** [comment]
@romanblanco now that you cut a release for 2.9.0, this is technically a breaking change. I was hoping it would make it in prior to the 'run_test!' options change. How would you like to handle this?

**Comment by romanblanco** [comment]
@mattpolito, my plan is to review and release this and other PRs in 2.10.0 before merging https://github.com/rswag/rswag/pull/574 and other future changes for 3.0.0.

I've tried to rebase your changes and run tests and everything is still passing.

**Comment by mattpolito** [comment]
@romanblanco It's not a breaking change in code but in usage. However after thinking about how ruby would handle the args vs options, I believe it will be fine after-all.

**Review by romanblanco — APPROVED** [review]
LGTM :+1: 

@mattpolito, @a-lavis, @ayunker, thanks for getting this done!

**Pull request merged**
